src/instruction.rs

```diff
@@ -0,0 +1,422 @@
+        {
+            v.push(instruction);
+        }
+        for instruction in BIT_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in BRANCHES_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in CMP_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in CPX_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in CPY_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in DEC_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in EOR_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in PROC_STATUS_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in INC_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in JMP_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in LDA_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in LDX_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in LDY_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in LSR_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in ORA_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in REGISTER_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in ROL_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in ROR_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in SBC_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in STA_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in STACK_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in STX_SET.iter()
+        {
+            v.push(instruction);
+        }
+        for instruction in STY_SET.iter()
+        {
+            v.push(instruction);
+        }
+        v
+    };
+
+    pub static ref INSTRUCTION_MAP : std::collections::HashMap<u8, &'static Instruction> = {
+        let mut m = std::collections::HashMap::<u8,&'static Instruction>::new();
+        for instruction in INSTRUCTIONS.iter() {
+            if let Some(_) = m.insert(instruction.opcode, instruction) {
+                panic!("INSTRUCTION_MAP : Error, opcode already defined adding instruction : {:#?}", instruction)
+            }
+        }
+        m
+    };
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{INSTRUCTION_MAP, Mode, Instruction};
+
+    fn get_lenght_from_mode(mode: &Mode) -> u8
+    {
```

Comment:
we need to use a formatter. For now I'd like to ask you to use rustfmt manually, but I guess we could setup the CI that does this automatically later on

Reply:
ok